gammapy/modeling/models/spectral.py

```diff
@@ -213,6 +213,45 @@ def evaluate_error(self, energy, epsilon=1e-4):
         """
         return self._propagate_error(epsilon=epsilon, fct=self, energy=energy)
 
+    @property
+    def pivot_energy(self):
+        """The pivot or decorrelation energy, for a given spectral model calculated numerically.
+        It is defined as the energy at which the correlation between the spectral parameters is minimized.
+
+        Returns
+        -------
+        pivot energy : `~astropy.units.Quantity`
+            The energy at which the statistical error in the computed flux is smallest.
+            If no minimum is found, NaN will be returned.
+        """
+
+        x_unit = self.reference.unit
+
+        def min_func(x):
+            """Function to minimize"""
+            x = np.exp(x)
+            dnde, dnde_error = self.evaluate_error(x * x_unit)
+            return dnde_error / dnde
+
+        bounds = [np.log(self.reference.value) - 3, np.log(self.reference.value) + 3]
+
+        std = np.std(min_func(x=np.linspace(bounds[0], bounds[1], 100)))
```

Comment:
does the choice of the number of steps in linespace can affect the precision ?
If yes it should be configurable.

Reply:
It does not